src/core.c/io_operators.rakumod

```diff
@@ -282,34 +282,77 @@ multi sub indir(IO() $path, &what, :$d = True, :$r, :$w, :$x) {
     }
 }
 
-proto sub chmod($, |) {*}
-multi sub chmod($mode, *@filenames) {
+proto sub chmod($, |) is revision-gated("6.c") {*}
+multi sub chmod($mode, Str $path) is revision-gated("6.c") { $path.IO.chmod($mode) }
+multi sub chmod($mode, IO::Path $path) is revision-gated("6.c") { $path.chmod($mode) }
+# In 6.e, we adjust the routine arguments to align with how chown is called
+# So we set up 6.e variants of the old signature that include deprecation messages
+# We also provide the new signature and make it available to 6.e+
+multi sub chmod($mode, Str $path) is revision-gated("6.e") {
+    DEPRECATED('chmod(Str $path, :$mode)', :what("old-fashioned chmod signature"));
+    $path.IO.chmod($mode)
+}
+multi sub chmod($mode, IO::Path $path) is revision-gated("6.e") {
+    DEPRECATED('chmod(IO::Path $path, :$mode)', :what("old-fashioned chmod signature"));
+    $path.chmod($mode)
+}
+multi sub chmod(Str $path, :$mode!) is revision-gated("6.e") { $path.IO.chmod($mode) }
+multi sub chmod(IO::Path $path, :$mode!) is revision-gated("6.e") { $path.chmod($mode) }
+multi sub chmod($mode, *@filenames) is revision-gated("6.c") {
+    my @ok;
+    for @filenames -> $file { @ok.push($file) if $file.IO.chmod($mode) }
+    @ok;
+}
+multi sub chmod($mode, *@filenames) is revision-gated("6.e") {
+    DEPRECATED('@paths.grep(*.IO.chmod)', :what("multi-path operation"));
     my @ok;
     for @filenames -> $file { @ok.push($file) if $file.IO.chmod($mode) }
     @ok;
 }
 
-proto sub chown($, |) {*}
-multi sub chown(*@filenames, :$uid, :$gid) {
+proto sub chown($, |) is revision-gated("6.c") {*}
+multi sub chown(Str $path, :$uid, :$gid) { $path.IO.chown(:$uid, :$gid) }
+multi sub chown(IO::Path $path, :$uid, :$gid) { $path.chown(:$uid, :$gid) }
+multi sub chown(*@filenames, :$uid, :$gid) is revision-gated("6.c") {
+    @filenames.grep: *.IO.chown(:$uid, :$gid)
+}
+multi sub chown(*@filenames, :$uid, :$gid) is revision-gated("6.e") {
+    DEPRECATED('@paths.grep(*.IO.chown)', :what("multi-path operation"));
     @filenames.grep: *.IO.chown(:$uid, :$gid)
 }
 
-proto sub unlink(|) {*}
+
+proto sub unlink(|) is revision-gated("6.c") {*}
 multi sub unlink() { "unlink()".no-zero-arg }
-multi sub unlink(*@filenames) {
+multi sub unlink(Str $path) { $path.IO.unlink }
+multi sub unlink(IO::Path $path) { $path.unlink }
+multi sub unlink(*@filenames) is revision-gated("6.c") {
+    my @ok;
+    for @filenames -> $file { @ok.push($file) if $file.IO.unlink }
+    @ok;
+}
+multi sub unlink(*@filenames) is revision-gated("6.e") {
+    DEPRECATED('@paths.grep(*.IO.unlink)', :what("multi-path operation"));
     my @ok;
     for @filenames -> $file { @ok.push($file) if $file.IO.unlink }
     @ok;
 }
 
-proto sub rmdir(|) {*}
+proto sub rmdir(|) is revision-gated("6.c") {*}
```

Comment:
Is the prototype supposed to be revision gated?

Reply:
It's my understanding that that is essential.

Reply:
If you don't put it on the prototype, the dispatcher will have no indication that it needs to engage revision gating amongst candidates.

Automatically adding this trait was not feasible and (imo) not advisable anyway. Sometimes our candidate lists span for hundreds of lines, so being explicit at the prototype declaration makes sense to me even if it were possible to avoid doing so.